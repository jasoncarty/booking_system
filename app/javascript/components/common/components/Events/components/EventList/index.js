import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Event from './../Event';

class EventList extends Component {
  static propTypes = {
    events: PropTypes.arrayOf(
      PropTypes.shape({})
    ).isRequired,
    adminSection: PropTypes.bool,
    attendances: PropTypes.arrayOf(PropTypes.number)
  }

  state = {
    events: []
  }

  componentDidMount() {
    this.setState({
      events: this.props.events
    });
  }

  handleDeleteEventSuccess = ({ data: { eventId }}) => {
    const events = this.state.events;
    this.setState({
      events: events.filter((event) => event.id.toString() !== eventId)
    });
  }

  handleDeleteEventError = (error) =>
    console.log(error, '----------------')

  getEvents = () => {
    const { events } = this.state;

    return events.map((event, index) => {
      const {
        name,
        description,
        starts_at,
        id,
        event_attendees: eventAttendees,
      } = event;

      const {
        adminSection,
        attendances
      } = this.props;

      return (
        <Event
          key={`${name}.${index}`}
          name={name}
          id={id}
          description={description}
          startsAt={starts_at}
          eventAttendees={eventAttendees}
          attendances={attendances}
          onDeleteSuccess={this.handleDeleteEventSuccess}
          onDeleteError={this.handleDeleteEventError}
          adminSection={adminSection}
        />
      );
    });
  }

  render() {
    return (
      <div id="event-container">
        <h1>Events</h1>
        <div className="spacer-25" />
        {this.getEvents()}
      </div>
    );
  }
}

export default EventList;
