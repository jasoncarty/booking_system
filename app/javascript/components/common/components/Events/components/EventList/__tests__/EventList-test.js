import React from 'react';
import { create } from 'react-test-renderer';

import EventList from './../index';

jest.mock('./../../Event', () => 'Event');

const defaultProps = {
  events: [
    { id: 1 },
    { id: 2 },
    { id: 3 }
  ]
};

describe('<EventList />', () => {
  it('matchesSnapshot with default props', () => {
    const element = create(<EventList {...defaultProps} />);
    expect(element).toMatchSnapshot();
  });

  it('deletes the event and changes state', () => {
    const element = create(<EventList {...defaultProps} />);
    const instance = element.getInstance();
    expect(instance.state.events).toEqual(defaultProps.events);

    instance.handleDeleteEventSuccess({
      data: {
        eventId: '1'
      }
    });

    expect(element).toMatchSnapshot();
    expect(instance.state.events).toEqual([
      { id: 2 },
      { id: 3 }
    ]);
  });
});
