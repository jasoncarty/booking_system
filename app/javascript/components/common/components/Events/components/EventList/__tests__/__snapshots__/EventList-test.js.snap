// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EventList /> deletes the event and changes state 1`] = `
<div
  id="event-container"
>
  <h1>
    Events
  </h1>
  <div
    className="spacer-25"
  />
  <Event
    id={2}
    onDeleteError={[Function]}
    onDeleteSuccess={[Function]}
  />
  <Event
    id={3}
    onDeleteError={[Function]}
    onDeleteSuccess={[Function]}
  />
</div>
`;

exports[`<EventList /> matchesSnapshot with default props 1`] = `
<div
  id="event-container"
>
  <h1>
    Events
  </h1>
  <div
    className="spacer-25"
  />
  <Event
    id={1}
    onDeleteError={[Function]}
    onDeleteSuccess={[Function]}
  />
  <Event
    id={2}
    onDeleteError={[Function]}
    onDeleteSuccess={[Function]}
  />
  <Event
    id={3}
    onDeleteError={[Function]}
    onDeleteSuccess={[Function]}
  />
</div>
`;
